### ISSUE TITLE
Extracting a key that is null in json returns a non-null value

### ISSUE THREAD

**Issue by kemiller** [duckdb/duckdb#6779]
### What happens?

When extracting from json, if a key in the json has a null value, it's returned as a non-null object in SQL.  This is not what I would expect at all, and breaks things like COALESCE.  

<img width="408" alt="image" src="https://user-images.githubusercontent.com/6359/226137555-5596c017-0511-4f34-a8f9-a1c9239f9de4.png">
<img width="460" alt="image" src="https://user-images.githubusercontent.com/6359/226137628-03b875e9-7075-44ee-8783-c55615efcfee.png">

### To Reproduce

Pretty easy to see what I mean by running this snippet:

```
select json_extract('{ "foo": null }', '$.foo') is null;
```

### OS:

macOS / m1

### DuckDB Version:

v0.7.1 b00b93f0b1

### DuckDB Client:

CLI & typescript

### Full Name:

Ken Miller

### Affiliation:

Watershed

### Have you tried this on the latest `master` branch?

- [x] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by kemiller** [comment]
Interestingly, converting to a struct and then extracting does the trick.  So we do have a workaround, but this doesn't seem like the right behavior to me.


![image](https://user-images.githubusercontent.com/6359/226141368-ac700ea3-8d6f-4d3d-b84d-8d3a3d6c30ff.png)

**Comment by kemiller** [comment]
It also works correctly if the key is just missing.

**Comment by Tishj** [comment]
`IS NULL` is likely not implemented correctly for the JSON type

**Comment by kemiller** [comment]
We noticed it because the output didn't behave correctly in COALESCE()

**Comment by Tishj** [comment]
I'm not that familiar with the JSON type, but internally it's just a string, so "null" isn't recognized as being NULL.
Which makes sense, but we probably need to add some special handling to mark these entries as being NULL

**Comment by kemiller** [comment]
Bump -- any news on this one?

**Comment by lnkuiper** [comment]
This is because `null` is valid `JSON`, and therefore is not DuckDB's `NULL`. This should probably be changed to go to our `NULL` once extracted.

**Comment by lnkuiper** [comment]
Closed via #7478

**Comment by kemiller** [comment]
Fantastic, thank you!

**Comment by dpmccabe** [comment]
Given that this fix was reverted by [#13481](https://github.com/duckdb/duckdb/pull/13481), should we use the `struct_extract` workaround again?

**Comment by lnkuiper** [comment]
@dpmccabe I don't think you need any struct functions to distinguish between a JSON `null` and a SQL `NULL`:
```sql
D select json_value('{"duck":null}', '$.duck');
┌───────────────────────────────────────┐
│ json_value('{"duck":null}', '$.duck') │
│                varchar                │
├───────────────────────────────────────┤
│                                       │
└───────────────────────────────────────┘
D select json_extract('{"duck":null}', '$.duck');
┌─────────────────────────────────────────┐
│ json_extract('{"duck":null}', '$.duck') │
│                  json                   │
├─────────────────────────────────────────┤
│ null                                    │
└─────────────────────────────────────────┘
```
The first one is a SQL `NULL`, while the second one is a JSON `null`.
